ios/RNSScreenStack.mm

```diff
@@ -57,6 +57,31 @@ - (UIViewController *)childViewControllerForStatusBarHidden
   return [self topViewController];
 }
 
+- (void)viewDidLayoutSubviews
+{
+  if ([self.topViewController isKindOfClass:[RNSScreen class]]) {
+    RNSScreen *screenController = (RNSScreen *)self.topViewController;
+
+    // If RNSScreen includes a navigation controller of type RNScreensNavigationController, it should not calculate
+    // header height, as it could have nested stack.
+    if (!(screenController.childViewControllers.count > 0 && screenController.hasAnyChildNavigators)) {
+      [(RNSScreen *)self.topViewController calculateHeaderHeightIsModal:NO];
+    }
+  }
+}
+
+- (void)dismissViewControllerAnimated:(BOOL)flag completion:(void (^)())completion
+{
+  [super dismissViewControllerAnimated:flag completion:completion];
+  BOOL wasModalPresented = ((RNSScreen *)self.presentedViewController).screenView.isModal;
+  // Check if presented modal was nested
+  BOOL isTopViewCtrNativeModal = ((RNSScreen *)self.topViewController).screenView.isPresentedAsNativeModal;
```

Comment:
Is it ever `false`? I thought `dismissViewControllerAnimated` is called for modals only.

Reply:
It is - that's why we're checking if topViewController is _presented as native modal_. If it is, we're getting its top view controller's child view controller and passing its height.

Also, I'm trying to recall why there was `wasModalPresented` introduced 🤔 It seems that it is indeed obsolete.

Reply:
Removed in https://github.com/software-mansion/react-native-screens/pull/1802/commits/94e52902d49a36e1f22a3059196898efc2a714ac, but I will check that further.